Source/WebCore/Modules/url-pattern/URLPattern.cpp

```diff
@@ -26,29 +26,247 @@
 #include "config.h"
 #include "URLPattern.h"
 
+#include "URLPatternCanonical.h"
 #include "URLPatternInit.h"
 #include "URLPatternOptions.h"
 #include "URLPatternResult.h"
 #include <wtf/RefCounted.h>
+#include <wtf/URL.h>
 
 namespace WebCore {
 
 WTF_MAKE_TZONE_OR_ISO_ALLOCATED_IMPL(URLPattern);
 
-Ref<URLPattern> URLPattern::create(URLPatternInput&&, String&& baseURL, URLPatternOptions&&)
+template<typename CharacterType>
+static String escapeStringForCharacters(std::span<const CharacterType> characters)
+{
+    static constexpr std::array<LChar, 9> escapeCharacters = { '+', '*', '?', ':', '(', ')', '\\', '{', '}' }; // NOLINT
```

Comment:
Would this build or does it really have to be `LChar`?
```
static constexpr std::array escapeCharacters = { '+', '*', '?', ':', '(', ')', '\\', '{', '}' };
```

Reply:
It does build but I thought we explicitly wanted it to be LChar? I'll change it back to that then for both escapeStringForCharacters and isValidIpv6Host

Reply:
I was having trouble with the linter complaining that the brackets needed spaces.